src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -643,6 +642,34 @@ public bool IsWebSocketRequest
         {
             get
             {
```

Comment:
I haven't changed much in this particular file. The new functionality added is mostly in websocketvalidate.http.managed.cs, acceptwebsocketasynccore and the other new referenced functions from that method. Some parsing of headers, I took from Windows and put it in Websocketvalidate.cs in common location.